cmd/kubeapps-apis/server/plugins.go

```diff
@@ -109,19 +128,43 @@ func registerPlugins(pluginPaths []string, grpcReg grpc.ServiceRegistrar, gwArgs
 }
 
 // registerGRPC finds and calls the required function for registering the plugin for the GRPC server.
-func registerGRPC(p *plugin.Plugin, pluginPath string, registrar grpc.ServiceRegistrar) error {
+func (s *pluginsServer) registerGRPC(p *plugin.Plugin, pluginDetail *plugins.Plugin, registrar grpc.ServiceRegistrar) error {
 	grpcRegFn, err := p.Lookup(grpcRegisterFunction)
 	if err != nil {
-		return fmt.Errorf("unable to lookup %q for %q: %w", grpcRegisterFunction, pluginPath, err)
+		return fmt.Errorf("unable to lookup %q for %v: %w", grpcRegisterFunction, pluginDetail, err)
 	}
-	type grpcRegisterFunctionType = func(grpc.ServiceRegistrar)
+	type grpcRegisterFunctionType = func(grpc.ServiceRegistrar) interface{}
 	grpcFn, ok := grpcRegFn.(grpcRegisterFunctionType)
 	if !ok {
-		var dummyFn grpcRegisterFunctionType = func(grpc.ServiceRegistrar) {}
-		return fmt.Errorf("unable to use %q in plugin %q due to mismatched signature.\nwant: %T\ngot: %T", grpcRegisterFunction, pluginPath, dummyFn, grpcRegFn)
+		var dummyFn grpcRegisterFunctionType = func(grpc.ServiceRegistrar) interface{} { return nil }
+		return fmt.Errorf("unable to use %q in plugin %v due to mismatched signature.\nwant: %T\ngot: %T", grpcRegisterFunction, pluginDetail, dummyFn, grpcRegFn)
 	}
-	grpcFn(registrar)
+	server := grpcFn(registrar)
 
+	return s.registerPluginsSatisfyingCoreAPIs(server, pluginDetail)
```

Comment:
This is the new piece that is called when registering a grpc function ^^^ , defined below.